providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/custom_ops.py

```diff
@@ -32,28 +39,51 @@ def fractional(data: dict, *args: JsonLogicArg) -> typing.Optional[str]:
         return None
 
     hash_ratio = abs(mmh3.hash(bucket_by)) / (2**31 - 1)
-    bucket = int(hash_ratio * 100)
+    bucket = hash_ratio * 100
 
+    total_weight = 0
+    fractions = []
     for arg in args:
-        if (
-            not isinstance(arg, (tuple, list))
-            or len(arg) != 2
-            or not isinstance(arg[0], str)
-            or not isinstance(arg[1], int)
-        ):
-            logger.error("Fractional variant weights must be (str, int) tuple")
-            return None
-    variant_weights: typing.Tuple[typing.Tuple[str, int]] = args  # type: ignore[assignment]
-
-    range_end = 0
-    for variant, weight in variant_weights:
-        range_end += weight
+        fraction = __parse_fraction(arg)
+        if fraction:
+            fractions.append(fraction)
+            total_weight += fraction.weight
+
+    range_end: float = 0
+    for fraction in fractions:
+        range_end += fraction.weight * 100 / total_weight
         if bucket < range_end:
-            return variant
+            return fraction.variant
 
     return None
 
 
+def __parse_fraction(arg: JsonLogicArg) -> typing.Optional[Fraction]:
```

Comment:
```suggestion
def _parse_fraction(arg: JsonLogicArg) -> typing.Optional[Fraction]:
```

Convention in python is to use a single leading underscore for private methods (as I think is the aim here), double leading underscores are used for name mangling and (afaik) are only relevant for class methods.